test/python/transpiler/test_basic_swap.py

```diff
@@ -334,6 +343,359 @@ def test_far_swap_with_gate_the_middle(self):
+        false_body.y(0)
+        qc.if_else((creg[0], 0), true_body, false_body, qreg, creg)
+        qc.barrier()
+        qc.measure(qreg, creg)
+
+        dag = circuit_to_dag(qc)
+        cdag = BasicSwap(coupling).run(dag)
+        cqc = dag_to_circuit(cdag)
+
+        sim = Aer.get_backend("aer_simulator")
+        in_results = sim.run(qc, shots=4096, seed_simulator=10).result().get_counts()
+        out_results = sim.run(cqc, shots=4096, seed_simulator=11).result().get_counts()
+        self.assertLess(hellinger_distance(in_results, out_results), 0.01)
+
+    @unittest.skipUnless(optionals.HAS_AER, "Qiskit Aer is required to run this test")
+    def test_controlflow_nested_outer_cnot(self):
+        """test swap with nested if else controlflow construct; swap in outer"""
+        num_qubits = 5
+        qreg = QuantumRegister(num_qubits, "q")
+        creg = ClassicalRegister(num_qubits)
+        coupling = CouplingMap([(i, i + 1) for i in range(num_qubits - 1)])
+        qc = QuantumCircuit(qreg, creg)
+        qc.h(0)
+        qc.x(1)
+        qc.measure(0, 0)
+        true_body = QuantumCircuit(qreg, creg)
+        true_body.cx(0, 2)
+        true_body.x(0)
+
+        for_body = QuantumCircuit(qreg, creg)
+        for_body.delay(10, 0)
+        for_body.cx(1, 3)
+        loop_parameter = None
+        true_body.for_loop(range(5), loop_parameter, for_body, qreg, creg)
+
+        false_body = QuantumCircuit(qreg, creg)
+        # false_body.cx(0, 4)
+        false_body.y(0)
+        qc.if_else((creg[0], 0), true_body, false_body, qreg, creg)
+        qc.barrier()
+        qc.measure(qreg, creg)
+
+        dag = circuit_to_dag(qc)
+        cdag = BasicSwap(coupling).run(dag)
+        cqc = dag_to_circuit(cdag)
+
+        sim = Aer.get_backend("aer_simulator")
+        in_results = sim.run(qc, shots=4096, seed_simulator=10).result().get_counts()
+        out_results = sim.run(cqc, shots=4096, seed_simulator=11).result().get_counts()
+        self.assertLess(hellinger_distance(in_results, out_results), 0.01)
+
+    @unittest.skipUnless(optionals.HAS_AER, "Qiskit Aer is required to run this test")
+    def test_controlflow_continue(self):
+        """test controlflow continue"""
+        num_qubits = 5
+        qreg = QuantumRegister(num_qubits, "q")
+        creg = ClassicalRegister(num_qubits)
+        coupling = CouplingMap([(i, i + 1) for i in range(num_qubits - 1)])
+        qc = QuantumCircuit(qreg, creg)
+        for_body = QuantumCircuit(qreg, creg)
+        for_body.cx(0, 2)
+        for_body.continue_loop()
+
+        loop_parameter = None
+        qc.for_loop(range(3), loop_parameter, for_body, qreg, creg)
+
+        dag = circuit_to_dag(qc)
+        cdag = BasicSwap(coupling).run(dag)
+        cqc = dag_to_circuit(cdag)
+
+        expected = QuantumCircuit(qreg, creg)
+        efor_body = QuantumCircuit(qreg, creg)
+        efor_body.swap(0, 1)
+        efor_body.cx(1, 2)
+        efor_body.continue_loop()
```

Comment:
This is an interesting case. Wouldn't the `continue` need to come after the final `swap` (to make sure the final layout remains consistent)?

Reply:
There is code in the stochastic_swap pr which addresses this. I'll try including here.